src/middlewared/middlewared/plugins/cloud_backup/restic.py

```diff
@@ -37,38 +40,80 @@ async def run_restic(job, cmd, env, stdin=None):
         stdout=subprocess.PIPE,
         stderr=subprocess.STDOUT,
     )
-    check_progress = asyncio.ensure_future(restic_check_progress(job, proc))
+    check_progress = asyncio.ensure_future(restic_check_progress(job, proc, cmd))
     cancelled_error = None
     try:
+        await proc.wait()
+    except asyncio.CancelledError as e:
+        cancelled_error = e
         try:
-            await proc.wait()
-        except asyncio.CancelledError as e:
-            cancelled_error = e
-            try:
-                await job.middleware.call("service.terminate_process", proc.pid)
-            except CallError as e:
-                job.middleware.logger.warning(f"Error terminating restic on cloud backup abort: {e!r}")
+            await job.middleware.call("service.terminate_process", proc.pid)
+        except CallError as e:
+            job.middleware.logger.warning(f"Error terminating restic on cloud backup abort: {e!r}")
     finally:
         await asyncio.wait_for(check_progress, None)
 
     if cancelled_error is not None:
         raise cancelled_error
     if proc.returncode != 0:
-        message = "".join(job.internal_data.get("messages", []))
+        message = "\n".join(job.internal_data.get("messages", []))
         if message and proc.returncode != 1:
             if not message.endswith("\n"):
                 message += "\n"
             message += f"restic failed with exit code {proc.returncode}"
         raise CallError(message)
 
 
-async def restic_check_progress(job, proc):
+async def restic_check_progress(job, proc, cmd: list[str]):
+    """Record progress of restic backup, restore, and forget commands.
+
+    Relevant documentation: https://restic.readthedocs.io/en/stable/075_scripting.html#json-output
+
+    """
+    if "forget" in cmd:
```

Comment:
We should make this an explicit flag, saying if we want to check progress or just want to copy the logs.

Reply:
We can, but we'll need to note that setting the flag on `restic forget` will cause this to fail since `forget` returns different JSON output than `backup` and `restore`.

Reply:
I understand, let's just not use methods like `"forget" in cmd` to determine what the output will be; let's use explicit flags instead